net/client.go

```diff
@@ -30,7 +30,20 @@ var (
 
 // pushLog creates a pushLog request and sends it to another node
 // over libp2p grpc connection
-func (s *server) pushLog(evt event.Update, pid peer.ID) error {
+func (s *server) pushLog(evt event.Update, pid peer.ID) (err error) {
+	defer func() {
+		if err != nil && !evt.IsRetry {
```

Comment:
thought: the failure event could be used to queue another retry instead of using a channel on the event

Reply:
> thought: the failure event could be used to queue another retry instead of using a channel on the event

The retry interval is set per peer and not per update. This is why the channel on the event is used to say if the retry was successful or not.